experiments/AMIP/moist_mpi_earth/coupler_utils/bcfile_reader.jl

```diff
@@ -192,3 +197,40 @@ function update_midmonth_data!(date, bcf_info)
 end
 
 next_date_in_file(bcfile_info) = bcfile_info.all_dates[bcfile_info.segment_idx[1] + Int(1)]
+
+# IO - daily
+"""
+    interpolate_midmonth_to_daily(date, bcf_info)
+
+Interpolates linearly between two `Fields` in the `bcf_info` struct, or returns the first Field if interpolation is switched off. 
+"""
+function interpolate_midmonth_to_daily(date, bcf_info)
+    @show bcf_info.segment_length[1]
+    if bcf_info.interpolate_daily[1] && bcf_info.segment_length[1] > FT(0)
+        segment_length = bcf_info.segment_length
+        segment_idx = bcf_info.segment_idx
+        all_dates = bcf_info.all_dates
+        monthly_fields = bcf_info.monthly_fields
+
+        return intepol.(
+            monthly_fields[1],
+            monthly_fields[2],
+            (date - all_dates[Int(segment_idx[1])]).value,
+            segment_length[1],
+            FT,
+        )
+    else
+        return bcf_info.monthly_fields[1]
+    end
+end
+
+"""
+    intepol(f_t1, f_t2, t, Δt, FT)
+
+Performs linear interpolation within a segment `Δt` = (t1 - t2), of fields `f_t1` and `f_t2`, with `t`(`f_t1`) = t1.
+"""
+function intepol(f1, f2, t, Δt, FT)
```

Comment:
you can instead do "interpol(f1::FT, f2::FT, t::FT, dt::FT) where {FT}" and not pass in the FT as well.